flocker/node/test/test_script.py

```diff
@@ -71,137 +81,325 @@ def deployer_factory_stub(**kw):
     return deployer
 
 
-class ZFSGenericAgentScriptTests(SynchronousTestCase):
+class DummyAgentService(PRecord):
+    reactor = field()
+    loop_service = field()
+    configuration = field()
+
+    @classmethod
+    def for_loop_service(cls, loop_service):
+        return lambda configuration: cls(
+            loop_service=loop_service,
+            configuration=configuration,
+        )
+
+    def get_api(self):
+        return None
+
+    def get_deployer(self, api):
+        return None
+
+    def get_loop_service(self, deployer):
+        return self.loop_service
+
+
+class DatasetServiceFactoryTests(SynchronousTestCase):
```

Comment:
:question: Perhaps the name should reflect the fact that it is testing with the ZFS backend only.